### PR TITLE
FED-3274 React 18 Prep

### PR DESCRIPTION
**Pull request by sydneyjodon-wk**
## Motivation
  

A couple test updates in order to get tests passing on React 18

## Changes
  

- Update context tests similar to react-dart since changedBits doesn't work on React 18
- Update ErrorBoundary prod tests to use `react_dom.render` to avoid RTL error about using `act` in prod builds
- Update analyzer plugin to fix analysis errors on Dart 3 (these were also happening in master)

## Review
_[See CONTRIBUTING.md][contributing-review-types] for more details on review types (+1 / QA +1 / +10) and code review process._

  

Please review: 

### QA Checklist
- [ ] Tests were updated and provide good coverage of the changeset and other affected code
- [ ] Manual testing was performed if needed
    - [ ] Steps from PR author: 
        
      - [ ] CI passes
      - [ ] CI passes on React 18 test #968 
    - [ ] Anything falling under manual testing criteria [outlined in CONTRIBUTING.md][contributing-manual-testing]

## Merge Checklist
While we perform many automated checks before auto-merging, some manual checks are needed:
- [ ] A Frontend Frameworks Design member has reviewed these changes
- [ ] There are no unaddressed comments _- this check can be automated if reviewers use the "Request Changes" feature_
- [ ] _For release PRs -_ Version metadata in Rosie comment is correct

[contributing-review-types]: https://github.com/Workiva/over_react/blob/master/CONTRIBUTING.md#review-types
[contributing-manual-testing]: https://github.com/Workiva/over_react/blob/master/CONTRIBUTING.md#manual-testing-criteria

### CODE REVIEW

File: tools/analyzer_plugin/lib/src/diagnostic/exhaustive_deps.dart [hunk 1]
```diff
@@ -1605,12 +1605,6 @@ Iterable<Union<Identifier, NamedType>> resolvedReferencesWithin(AstNode node) sy
   }
 }
 
-extension on NamedType {
```
Comment (sydneyjodon-wk) [inline tools/analyzer_plugin/lib/src/diagnostic/exhaustive_deps.dart:1608]: These analyzer_plugin changes fix the failing analysis on Dart 3 in CI

File: test/over_react/component/error_boundary/shared_stack_tests.dart [hunk 2]
```diff
@@ -27,17 +29,15 @@ void sharedErrorBoundaryStackTests() {
...
-            capturedInfos.add(info);
-          })(element));
-        // Use prints as an easy way to swallow `print` calls and
-        // prevent RTL from forwarding console errors to the test output,
-        // since React error boundary logging is pretty noisy.
-        // TODO instead, disable logging in this rtl.render call once that option is available: FED-1641
-      }, prints(anything));
+      final mountNode = DivElement();
+      // Use react_dom.render instead of RTL to avoid errors on React 18 about
+      // `act` being used in prod builds.
```
Comment (aaronlademann-wf) [inline test/over_react/component/error_boundary/shared_stack_tests.dart:34]: Is this being addressed in a separate RTL-related React 18 prep ticket?

File: tools/analyzer_plugin/lib/src/diagnostic/exhaustive_deps.dart (hunk 1 above)
Comment (aaronlademann-wf) [reply tools/analyzer_plugin/lib/src/diagnostic/exhaustive_deps.dart:1608]: Should we bump the SDK lower bound to `2.19.0` in the analyzer plugin pubspec?

**Review by aaronlademann-wf — APPROVED** [review]
Couple of questions, but otherwise LGTM

File: tools/analyzer_plugin/lib/src/diagnostic/exhaustive_deps.dart (hunk 1 above)
Comment (sydneyjodon-wk) [reply tools/analyzer_plugin/lib/src/diagnostic/exhaustive_deps.dart:1608]: Yep I can do that!

File: test/over_react/component/error_boundary/shared_stack_tests.dart (hunk 2 above)
Comment (sydneyjodon-wk) [reply test/over_react/component/error_boundary/shared_stack_tests.dart:34]: @greglittlefield-wf is working on RTL prep - I'm not sure if this is something that would be addressed or if it's more like RTL shouldn't be used in prod builds in general and this is an edge case where a test is purposely trying to use a prod build?

**Review by aaronlademann-wf — DISMISSED** [review]

File: test/over_react/component/error_boundary/shared_stack_tests.dart (hunk 2 above)
Comment (greglittlefield-wf) [reply test/over_react/component/error_boundary/shared_stack_tests.dart:34]: > it's more like RTL shouldn't be used in prod builds in general and this is an edge case where a test is purposely trying to use a prod build?

Yeah I'd lean toward that, and don't think we need to try to address anything in RTL unless this comes up again

File: tools/analyzer_plugin/pubspec.yaml [hunk 3]
```diff
@@ -4,7 +4,7 @@ publish_to: none
 description: Dart analyzer plugin for OverReact
 repository: https://github.com/Workiva/over_react/tree/master/tools/analyzer_plugin
 environment:
-  sdk: '>=2.17.0 <3.0.0'
+  sdk: '>=2.19.0 <3.0.0'
 dependencies:
   analyzer: ^5.1.0
```
Comment (greglittlefield-wf) [inline tools/analyzer_plugin/pubspec.yaml:9]: The SDK constraints prevent incompatible versions from being pulled in, but since we're relying on APIs in newer analyzer versions—see above comment from that extension that was removed:
> // NamedType.element is added in analyzer 5.11.0…

—can we also update the constraint here to reflect that?
```suggestion
  analyzer: ^5.11.0
```

**Review by greglittlefield-wf — CHANGES_REQUESTED** [review]
Just one small comment; thanks for fixing that analyzer plugin issue!

File: tools/analyzer_plugin/pubspec.yaml (hunk 3 above)
Comment (sydneyjodon-wk) [reply tools/analyzer_plugin/pubspec.yaml:9]: Yep! Thanks for catching that!

**Review by greglittlefield-wf — APPROVED** [review]
- Changes look good
- CI passes with React 18 pulled in
- Assuming CI passes again on this PR on the latest commit

+10

**Review by aaronlademann-wf — APPROVED** [review]

**Comment by sydneyjodon-wk** [comment]
@Workiva/release-management-p

**Review by rmconsole-wf — APPROVED** [review]
+1 from RM
